Frontend/ui-common/src/Engineering/Helpers/JBridge.js

```diff
@@ -600,27 +600,16 @@ export const updateRouteMarker = function (data) {
   }
 }
 
-export const updateRoute = function (data) {
-  return function (destMarker) {
-    return function (eta) {
-      return function (srcMarker){
-        return function (specialLocation) {
-          return function () {
-            if (window.JBridge.updateRoute) {
-              var json = JSON.stringify(data);
-              try{
-                console.log("I AM HERE ------------------ IN UPDATE ROUTE");
-                return window.JBridge.updateRoute(json, destMarker, eta, srcMarker, JSON.stringify(specialLocation));
-              }catch (err){
-                console.log("Catch error" + err);
-                return window.JBridge.updateRoute(json, destMarker, eta, JSON.stringify(specialLocation));
-              }
-            }
-          }
-        };
-      };
-    };
-  };
+export const updateRoute = function (data, destMarker, eta, srcMarker, specialLocation, zoomLevel) {
+    if (window.JBridge.updateRoute) {
+      var json = JSON.stringify(data);
+    try{
+        var payload = {"data":json, "destMarker": destMarker, "eta": eta, "srcMarker": srcMarker, "specialLocation": JSON.stringify(specialLocation), "zoomLevel": zoomLevel }
+        return window.JBridge.updateRoute(JSON.stringify(payload));
```

Comment:
create config in PureScript and modify the config while calling this function. refer  `startLottieProcess` function